sinks/attribution/attribution.go

```diff
@@ -0,0 +1,346 @@
+package attribution
+
+import (
+	"bytes"
+	"compress/gzip"
+	"context"
+	"encoding/csv"
+	"errors"
+	"fmt"
+	"io"
+	"sort"
+	"strings"
+	"time"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/credentials"
+	"github.com/aws/aws-sdk-go/aws/session"
+	"github.com/aws/aws-sdk-go/service/s3"
+	"github.com/aws/aws-sdk-go/service/s3/s3iface"
+	"github.com/axiomhq/hyperloglog"
+	"github.com/segmentio/fasthash/fnv1a"
+	"github.com/sirupsen/logrus"
+
+	veneur "github.com/stripe/veneur/v14"
+	"github.com/stripe/veneur/v14/samplers"
+	"github.com/stripe/veneur/v14/sinks"
+	"github.com/stripe/veneur/v14/ssf"
+	"github.com/stripe/veneur/v14/trace"
+	"github.com/stripe/veneur/v14/util"
+)
+
+// AttributionSinkConfig represents the YAML options that can be used to
+// configure an instance of AttributionSink
+type AttributionSinkConfig struct {
+	AWSAccessKeyID     util.StringSecret `yaml:"aws_access_key_id"`
+	AWSRegion          string            `yaml:"aws_region"`
+	AWSSecretAccessKey util.StringSecret `yaml:"aws_secret_access_key"`
+	HostnameTag        string            `yaml:"hostname_tag"`
+	SchemaVersion      int               `yaml:"schema_version"`
+	OwnerKey           string            `yaml:"owner_key"`
+	S3Bucket           string            `yaml:"s3_bucket"`
+	S3KeyPrefix        string            `yaml:"s3_key_prefix"`
+	VeneurInstanceID   string            `yaml:"veneur_instance_id"`
+}
+
+// ParseConfig decodes the map config for an S3 sink into an S3SinkConfig
```

Comment:
nit: did you mean `AttributionSinkConfig` instead of `S3SinkConfig`?

Reply:
yes, thanks